From d63d9f84a1a69ac10daffe5bf94e61c093bfe610 Mon Sep 17 00:00:00 2001
From: Peter Wu <peter@lekensteyn.nl>
Date: Sat, 10 Jan 2015 23:39:43 +0100
Subject: [PATCH] daemon: fix build with newer libzrtpcpp

When compiling with libzrtpcpp 4.3.1 on Arch Linux (which uses
https://github.com/wernerd/ZRTPCPP), the build fails because ZRTP_CFLAGS
is not passed to the cpp files and as such the -I${includedir/libzrtpcpp
option is also missing.

The next attempt was to rename CFLAGS to CPPFLAGS (to capture both
CFLAGS/CXXFLAGS), but this did not solve the zrtp inclusion issue.

It turns out that the zrtpccrtp.h header (as included by
audio/audiortp/audio_zrtp_session.h) depends on
-I${includedir}/libzrtpcpp
(https://github.com/wernerd/ZRTPCPP/issues/20), but this flag is not
passed to the correct files as the parent ..._la_CFLAGS are not
inherited by the subdirs.

To fix this, add ZRTPCPP_CFLAGS to the direct users of the
audio/audiortp/audio_zrtp_session.h header.

Refs #51388

Change-Id: If42411d86137a9e2d647867233783f3b451b9239
---
 daemon/src/Makefile.am                | 1 -
 daemon/src/audio/audiortp/Makefile.am | 2 ++
 daemon/src/client/Makefile.am         | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/daemon/src/Makefile.am b/daemon/src/Makefile.am
index 08d722c..b8c9581 100644
--- a/daemon/src/Makefile.am
+++ b/daemon/src/Makefile.am
@@ -76,7 +76,6 @@ libsflphone_la_LDFLAGS += @PCRE_LIBS@
 endif
 
 libsflphone_la_CFLAGS = \
-		@ZRTPCPP_CFLAGS@ \
 		$(PJSIP_CFLAGS) \
 		@CCRTP_CFLAGS@ \
 		@ALSA_CFLAGS@ \
diff --git a/daemon/src/audio/audiortp/Makefile.am b/daemon/src/audio/audiortp/Makefile.am
index 642f365..4a0e5f0 100644
--- a/daemon/src/audio/audiortp/Makefile.am
+++ b/daemon/src/audio/audiortp/Makefile.am
@@ -4,6 +4,8 @@ noinst_LTLIBRARIES = libaudiortp.la
 
 if BUILD_ZRTP
 SFL_ZRTP_SRC=audio_zrtp_session.h audio_zrtp_session.cpp zrtp_session_callback.cpp zrtp_session_callback.h
+libaudiortp_la_CXXFLAGS = \
+		@ZRTPCPP_CFLAGS@
 endif
 
 libaudiortp_la_SOURCES = \
diff --git a/daemon/src/client/Makefile.am b/daemon/src/client/Makefile.am
index 45b0783..72ab71b 100644
--- a/daemon/src/client/Makefile.am
+++ b/daemon/src/client/Makefile.am
@@ -37,5 +37,6 @@ libclient_la_SOURCES = callmanager.cpp           \
                        $(STUB_SRC)
 
 libclient_la_CXXFLAGS = -I./ \
+                        @ZRTPCPP_CFLAGS@ \
                         -DPREFIX=\"$(prefix)\" \
                         -DPROGSHAREDIR=\"${datadir}/sflphone\"
-- 
2.2.1

